src/main/java/frc/robot/subsystems/ball/shooter/ShooterImpl.java

```diff
@@ -21,22 +26,29 @@ private double getShooterRPM() {
         return ((this.shooterEncoder.getRate() / 128) * 60);
     }
     
-    public void shoot(boolean shooting) {
-        if (shooting){
-            if (getShooterRPM() > 4000){
-                this.shooterController.set(1.0d);
-            }
+    public void runShooter() {
+        if (getShooterRPM() < 4000){
```

Comment:
This will be something you want to spend a lot of time tuning, so I'd look at using a SmartTuner here rather than hardcoding it in this if condition.